linera-views/src/memory.rs

```diff
@@ -193,84 +201,109 @@ impl WritableKeyValueStore<MemoryStoreError> for MemoryStore {
         Ok(())
     }
 
-    async fn clear_journal(&self, _base_key: &[u8]) -> Result<(), MemoryStoreError> {
+    async fn clear_journal(&self) -> Result<(), MemoryStoreError> {
         Ok(())
     }
 }
 
 impl MemoryStore {
     fn sync_connect(
-        namespace_memory_store: &NamespaceMemoryStore,
+        memory_stores: &mut MemoryStores,
         config: &MemoryStoreConfig,
         namespace: &str,
+        root_key: &[u8],
         kill_on_drop: bool,
     ) -> Result<Self, MemoryStoreError> {
         let max_stream_queries = config.common_config.max_stream_queries;
-        let namespace = namespace.to_string();
-        let store = namespace_memory_store
-            .get(&namespace)
-            .ok_or(MemoryStoreError::NotExistentNamespace)?;
+        ensure!(
+            memory_stores.namespaces.contains(namespace),
+            MemoryStoreError::NotExistentNamespace
```

Comment:
Would `NonExistentNamespace` or `NamespaceNotFound` be better?

Reply:
Ok, I used `NamespaceNotFound`.